platform/security/security-migratable/src/main/java/org/codice/ddf/security/migratable/impl/SecurityMigratable.java

```diff
@@ -39,74 +31,83 @@
             Paths.get("etc", "ws-security", "issuer", "encryption.properties"),
             Paths.get("etc", "ws-security", "issuer", "signature.properties"));
 
     private static final String CRL_PROP_KEY = "org.apache.ws.security.crypto.merlin.x509crl.file";
 
-    private final MigratableUtil migratableUtil;
-
-    public SecurityMigratable(@NotNull DescribableBean info,
-            @NotNull MigratableUtil migratableUtil) {
-
-        super(info);
-
-        notNull(migratableUtil);
-        this.migratableUtil = migratableUtil;
+    @Override
+    public String getVersion() {
+        return SecurityMigratable.VERSION;
     }
 
-    public MigrationMetadata export(Path exportPath) throws MigrationException {
-        Collection<MigrationWarning> migrationWarnings = new ArrayList<>();
-        exportCrlFiles(exportPath, migrationWarnings);
-        exportPdpDirectory(exportPath, migrationWarnings);
-        return new MigrationMetadata(migrationWarnings);
+    @Override
+    public String getId() {
+        return "ddf.security";
     }
 
-    private void exportCrlFiles(Path exportDirectory,
-            Collection<MigrationWarning> migrationWarnings) throws MigrationException {
-        for (Path propertiesPath : PROPERTIES_FILES) {
-            exportCrlFile(propertiesPath, exportDirectory, migrationWarnings);
-        }
+    @Override
+    public String getTitle() {
+        return "Security Migration";
     }
 
-    private void exportCrlFile(Path propertiesPath, Path exportDirectory,
-            Collection<MigrationWarning> migrationWarnings) throws MigrationException {
-        LOGGER.debug("Exporting CRL from property [{}] in file [{}]...",
-                CRL_PROP_KEY,
-                propertiesPath.toString());
-        String crlPathStr = migratableUtil.getJavaPropertyValue(propertiesPath, CRL_PROP_KEY);
-
-        if (crlPathStr == null) {
-            return;
-        }
+    @Override
+    public String getDescription() {
+        return "Exports Security system files";
+    }
 
-        if (StringUtils.isWhitespace(crlPathStr)) {
-            String error = String.format(
-                    "Failed to export CRL. No CRL path found in file [%s]. Property [%s] from properties file [%s] has a blank value.",
-                    propertiesPath,
-                    CRL_PROP_KEY,
-                    propertiesPath);
-            throw new ExportMigrationException(error);
-        }
+    @Override
+    public String getOrganization() {
+        return "Codice";
+    }
 
-        Path crlPath = Paths.get(crlPathStr);
-        migratableUtil.copyFile(crlPath, exportDirectory, migrationWarnings);
+    @Override
+    public void doExport(ExportMigrationContext context) {
+        SecurityMigratable.PROPERTIES_FILES.stream()
+                .map(context::getEntry)
+                .peek(me -> LOGGER.debug("Exporting CRL from property [{}] in file [{}]...",
```

Comment:
✏️ I think an iteration through the list might work better here. As it is, you're over-logging because you'll log even for files that don't have the referenced property.

Reply:
I actually wanted to log all entries on purpose as they are attempted. This is what was done before.